include/zephyr/bluetooth/conn.h

```diff
@@ -1173,8 +1173,11 @@ struct bt_conn_cb {
 	 * @ref bt_conn_unref which is used by the BT stack. Making
 	 * Bluetooth API calls in this context is error-prone and strongly
 	 * discouraged.
+	 *
+	 * @param conn Connection object that was returned. Can ONLY be used
+	 *             for comparison purposes, not API calls.
 	 */
-	void (*recycled)(void);
+	void (*recycled)(const void *conn);
```

Comment:
Not sure I understand why this would use `const void *conn` instead of `const bt_conn *conn`. Can you elaborate?

Reply:
I see. I understand the intent then, but I do not think it's a good solution. 

If the intent is to compare the `void *` to another `bt_conn *`, then you also imply that the `void *` is an `bt_conn *` anyways, and we should expect application to simply cast it anyways (some static analyzers or compilers may even require that a pointer comparison use the same type, and thus request the `void *` to be cast anyhow). 

If we want to supply the connection pointer, I do think we should just be transparent and make it a `const bt_conn *conn`.